synapseclient/core/retry.py

```diff
@@ -80,44 +113,23 @@ def foo(a, b, c): return [a, b, c]
         # Try making the call
         try:
             response = function()
+            if retry_evaluator:
+                retry = retry_evaluator(response)
+        except SynapseThrottling as throttling_ex:
+            retry = True
+            wait = throttling_ex.wait
         except Exception as ex:
             exc = ex
             exc_info = sys.exc_info()
             logger.debug("calling %s resulted in an Exception" % function)
             if hasattr(ex, 'response'):
                 response = ex.response
 
-        # Check if we got a retry-able error
-        if response is not None and hasattr(response, 'status_code'):
-            if (
-                (expected_status_codes and response.status_code not in expected_status_codes) or
-                (retry_status_codes and response.status_code in retry_status_codes)
-            ):
-                response_message = _get_message(response)
-                retry = True
-                logger.debug("retrying on status code: %s" % str(response.status_code))
-                # TODO: this was originally printed regardless of 'verbose' was that behavior correct?
-                logger.debug(str(response_message))
-                if (response.status_code == 429) and (wait > 10):
-                    logger.warning('%s...\n' % response_message)
-                    logger.warning('Retrying in %i seconds' % wait)
-
-            elif response.status_code not in range(200, 299):
-                # For all other non 200 messages look for retryable errors in the body or reason field
-                response_message = _get_message(response)
-                if any([msg.lower() in response_message.lower() for msg in retry_errors]):
-                    retry = True
-                    logger.debug('retrying %s' % response_message)
-                # special case for message throttling
-                elif 'Please slow down.  You may send a maximum of 10 message' in response:
-                    retry = True
-                    wait = 16
-                    logger.debug("retrying " + response_message)
-
         # Check if we got a retry-able exception
         if exc is not None:
             if (exc.__class__.__name__ in retry_exceptions or
                     exc.__class__ in retry_exceptions or
+                    not retry_errors or
```

Comment:
Is this to check if retry_errors is None? If so should it be (retry_errors and any([msg.lower() in str(exc_info[1]).lower() for msg in retry_errors])?

And if we need to check that for None should we also check the other fields?

Reply:
I want to implement the logic that if users didn't pass any exception as the argument, I will default retry cuz we default to accept all exception